src/test/java/uk/gov/pay/api/resources/PaymentsResourceCreatePaymentTest.java

```diff
@@ -76,22 +84,24 @@ public void setup() {
                 connectorUriGenerator,
                 getPaymentService,
                 capturePaymentService,
-                cancelPaymentService);
+                cancelPaymentService,
+                paymentRequestValidator);
         when(publicApiUriGenerator.getPaymentURI(anyString())).thenReturn(URI.create(paymentUri));
     }
 
     @Test
     public void createNewPayment_withCardPayment_invokesCreatePaymentService() {
         final Account account = new Account("foo", TokenPaymentType.CARD);
-        final ValidCreatePaymentRequest createPaymentRequest = new ValidCreatePaymentRequest(CreatePaymentRequest.builder()
+        final CreatePaymentRequest createPaymentRequest = CreatePaymentRequest.builder()
                 .amount(100)
                 .returnUrl("https://somewhere.test")
                 .reference("my_ref")
                 .description("New Passport")
-                .build());
+                .build();
+
         PaymentWithAllLinks injectedResponse = aSuccessfullyCreatedPayment();
 
-        when(createPaymentService.create(account, createPaymentRequest)).thenReturn(injectedResponse);
+        when(createPaymentService.create(eq(account), ArgumentMatchers.any())).thenReturn(injectedResponse);
```

Comment:
Can we really not do better than `any()`? (Even `any(CreatePaymentRequest.class)` would be better!)

Feel free to add an `equals(…)` method to `CreatePaymentRequest` if it makes this sort of matching easier.

Reply:
Yes good point. I've added an equals method to `ValidCreatePaymentRequest` and used 
`        final ValidCreatePaymentRequest validCreatePaymentRequest = new ValidCreatePaymentRequest(createPaymentRequest);
`
then....
`eq(validCreatePaymentRequest)`